File: leetcode/55. Jump Game/main.java
// Solved by @kitanoyoru
// https://leetcode.com/problems/jump-game/

class Solution {
  public boolean canJump(int[] nums) {
    int n = nums.length;

    boolean[] dp = new boolean[n];
    dp[0] = true;

    for (int i = 1; i < n; i++) {
      for (int j = i - 1; j >= 0; j--) {
        if (dp[j] && j + nums[j] >= i) {
          dp[i] = true;
          break;
        }
      }
    }

    return dp[n - 1];
  }
}

File: leetcode/55. Jump Game/main.ts
// Solved by @kitanoyoru
// https://leetcode.com/problems/jump-game/

const canJump = (nums: number[]): boolean => {
  const n = nums.length

  const dp = new Array<boolean>(n)
  dp[0] = true

  for (let i = 0; i < n; i++) {
    for (let j = i - 1; j >= 0; j--) {
      if (dp[j] && j + nums[j] >= i) {
        dp[i] = true
        break
      }
    }
  }

  return dp[n - 1]
}
